### ISSUE TITLE
`no-unused-vars` doesn't work in JSX and TSX files

### ISSUE THREAD

**Issue by Skalman** [denoland/deno_lint#1054]
### Lint name

no-unused-vars

### Code Snippet

```ts
const foo = 42;
```

### Expected Result

> `foo` is never used

`no-unused-vars` rule is triggered regardless of the file type (extension).

### Actual Result

No lint error

### Additional Info

This is an intentional duplicate of #712. That issue was closed, since the rule intentionally skips JSX files to avoid false positives around the factory functions.

However, `deno_lint` should have access to `deno.jsonc` and perhaps `tsconfig.json`, and should be able to figure out what the factory imports are.

### Version

deno 1.23.0 (release, x86_64-unknown-linux-gnu)
v8 10.4.132.5
typescript 4.7.2

**Comment by rshurts** [comment]
Just checking out deno using fresh today and this tripped me up when I set up my editor with deno, I thought I had done something wrong as this was the error I introduced to test it out in editor linting.

I agree that skipping jsx/tsx doesn't seem to make much sense and makes me wonder what other may be skipped (now or in the future).
